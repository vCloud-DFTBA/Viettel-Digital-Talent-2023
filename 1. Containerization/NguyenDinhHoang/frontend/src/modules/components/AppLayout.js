import AppFrame from "./AppFrame";
import GlobalStyles from '@mui/material/GlobalStyles';
import { styled } from '@mui/material/styles';
import AppContainer from "./AppContainer";

const Main = styled('main', {
  shouldForwardProp: (prop) => prop !== 'disableToc',
})(({ disableToc, theme }) => ({
  display: 'flex',
  width: '100%',
  ...(disableToc && {
    [theme.breakpoints.up('lg')]: {
      marginRight: '5%',
    },
  }),
  [theme.breakpoints.up('lg')]: {
    width: 'calc(100% - var(--MuiDocs-navDrawer-width))',
  },
}));

const StyledAppContainer = styled(AppContainer, {
  shouldForwardProp: (prop) => prop !== 'disableAd' && prop !== 'disableToc',
})(({ disableAd, disableToc, theme }) => {
  return {
    position: 'relative',
    ...(!disableAd && {
      '&& .description': {
        marginBottom: 198,
      },
      '&& .description.ad': {
        marginBottom: 40,
      },
    }),
    ...(!disableToc && {
      [theme.breakpoints.up('sm')]: {
        width: 'calc(100% - var(--MuiDocs-toc-width))',
      },
      [theme.breakpoints.up('lg')]: {
        paddingLeft: '60px',
        paddingRight: '60px',
      },
    }),
  };
});

export function AppLayout(props) {
  
  const {
    children,
    disableAd = true,
    disableToc = true,
  } = props;

  return (
    <AppFrame>
      <GlobalStyles
        styles={{
          ':root': {
            '--MuiDocs-navDrawer-width': '300px',
            '--MuiDocs-toc-width': '240px',
          },
        }}
      />
      <Main disableToc={disableToc}>
        <StyledAppContainer disableAd={disableAd} disableToc={disableToc}>
          {children}
        </StyledAppContainer>
      </Main>
    </AppFrame>
  );
}