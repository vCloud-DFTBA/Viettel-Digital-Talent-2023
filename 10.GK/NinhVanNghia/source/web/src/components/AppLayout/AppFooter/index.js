import React from "react";


function AppFooter() {
    return (
        <div className="d-flex flex-row justify-content-between align-items-center p-5 mt-auto">
            <span className="text-muted">@Viettel Solution</span>
        </div>
    )
}


export default AppFooter
