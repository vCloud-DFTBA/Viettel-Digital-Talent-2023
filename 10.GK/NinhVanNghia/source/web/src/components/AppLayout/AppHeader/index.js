import React from "react";


function AppHeader() {
    return (
        <div className="container d-flex flex-row justify-content-between align-items-center py-5">
            <h2>VDT APP</h2>
        </div>
    )
}


export default AppHeader
