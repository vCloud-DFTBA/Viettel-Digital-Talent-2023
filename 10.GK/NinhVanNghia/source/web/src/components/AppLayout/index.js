import React from "react";

import AppHeader from "./AppHeader";
import AppFooter from "./AppFooter";


function AppLayout({children}) {
    return (
        <div className="App vw-100">
            <div className="page-wrapper d-flex flex-column flew-grow-1 min-vh-100">
                <AppHeader/>
                {children}
                <AppFooter/>
            </div>
        </div>
    )
}


export default AppLayout
