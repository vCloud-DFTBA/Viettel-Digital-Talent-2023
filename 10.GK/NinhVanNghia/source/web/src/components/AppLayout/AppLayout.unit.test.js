import React from 'react';
import renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';

import AppLayout from './index.js';


describe("<AppLayout/>", () => {
    test("It should render", () => {
        expect(renderer.create(<AppLayout/>).toJSON()).toMatchSnapshot()
    })

    test("It should render child component", () => {
        render(
            <AppLayout>
                some html
            </AppLayout>
        )
        expect(screen.getByText('some html')).toBeDefined()
    })
})
