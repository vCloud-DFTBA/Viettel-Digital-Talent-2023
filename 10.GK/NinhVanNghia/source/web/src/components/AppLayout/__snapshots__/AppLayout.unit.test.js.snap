// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AppLayout/> It should render 1`] = `
<div
  className="App vw-100"
>
  <div
    className="page-wrapper d-flex flex-column flew-grow-1 min-vh-100"
  >
    <div
      className="container d-flex flex-row justify-content-between align-items-center py-5"
    >
      <h2>
        VDT APP
      </h2>
    </div>
    <div
      className="d-flex flex-row justify-content-between align-items-center p-5 mt-auto"
    >
      <span
        className="text-muted"
      >
        @Viettel Solution
      </span>
    </div>
  </div>
</div>
`;
